pds_doi_service/core/outputs/osti/osti_web_client.py

```diff
@@ -0,0 +1,181 @@
+
+    def submit_content(self, payload, content_type=CONTENT_TYPE_XML):
+        """
+        Submits a payload to the OSTI DOI service via the POST action.
+
+        The action taken by the service is determined by the contents of the
+        payload.
+
+        The OSTI endpoint URL and authentication credentials for the request
+        are automatically pulled from the configuration file.
+
+        Parameters
+        ----------
+        payload : str
+            Payload to submit to the OSTI DOI service. Should correspond to
+            an OSTI-format label file containing a single DOI record.
+        content_type : str
+            The content type to specify the format of the payload, as well as
+            the format of the response from OSTI. Currently, 'xml' and 'json'
+            are supported.
+
+        Returns
+        -------
+        doi : Doi
+            Doi object parsed from the response label from OSTI.
+        response_text : str
+            Body of the response label from OSTI.
+
+        """
+        config = self._config_util.get_config()
+
+        response_text = super()._submit_content(
+            payload,
+            url=config.get('OSTI', 'url'),
+            username=config.get('OSTI', 'user'),
+            password=config.get('OSTI', 'password'),
+            content_type=content_type
+        )
+
+        # Re-use the parse functions from DOIOstiWebParser class to get the
+        # list of Doi objects to return
+        dois, _ = self._web_parser.parse_dois_from_label(response_text, content_type)
+
+        return dois[0], response_text
+
+    def query_doi(self, query, content_type=CONTENT_TYPE_XML):
+        """
+        Queries the status of a DOI from the OSTI server and returns the
+        response text.
+
+        The format of i_url is: https://www.osti.gov/iad2test/api/records/
+        and will be appended by fields in query_dict:
+
+            if query = {'id':14108,'status':'Error'}
+                then https://www.osti.gov/iad2test/api/records?id=14108&status=Error
+
+            if query = {'id'=1327397,'status'='Registered'}
+                then https://www.osti.gov/iad2test/api/records?id=1327397&status=Registered
+        """
+        config = self._config_util.get_config()
+
+        if content_type not in self._content_type_map:
+            raise ValueError('Invalid content type requested, must be one of '
+                             f'{",".join(list(self._content_type_map.keys()))}')
+
+        auth = HTTPBasicAuth(
+            config.get('OSTI', 'user'), config.get('OSTI', 'password')
+        )
+
+        headers = {
+            'Accept': self._content_type_map[content_type],
+            'Content-Type': self._content_type_map[content_type]
+        }
+
+        # OSTI server requires 'rows' field to know how many max rows to fetch at once.
+        initial_payload = {'rows': self.MAX_TOTAL_ROWS_RETRIEVE}
```

Comment:
I get that `MAX_TOTAL_ROWS_RETRIEVE` is a gazillion but are there practical limits to what OSTI can give back even if you say a gazillion is fine? Does the API support pagination?

Reply:
Not sure about this one either, this constant pre-dates my time on PDS.